Marlin/src/feature/powerloss.cpp

```diff
@@ -409,15 +407,15 @@ void PrintJobRecovery::resume() {
       gcode.process_subcommands_now(cmd);
     }
 
-    // Home XY with no Z raise, and also home Z here if Z isn't homing down below.
+    // Home XY with no Z raise
     gcode.process_subcommands_now_P(PSTR("G28R0XY")); // No raise during G28
 
   #endif
 
   #if HOMING_Z_DOWN
     // Move to a safe XY position and home Z while avoiding the print.
     constexpr xy_pos_t p = POWER_LOSS_ZHOME_POS;
-    sprintf_P(cmd, PSTR("G1X%sY%sF1000\nG28Z"), dtostrf(p.x, 1, 3, str_1), dtostrf(p.y, 1, 3, str_2));
+    sprintf_P(cmd, PSTR("G1X%sY%sF1000\nG28HZ"), dtostrf(p.x, 1, 3, str_1), dtostrf(p.y, 1, 3, str_2));
```

Comment:
This is causing the printer with z probe to home outside the print area because `H` doesn't respect the `NOZZLE_TO_PROBE_OFFSET`

Reply:
Well, the `POWER_LOSS_ZHOME_POS` is meant to be set w/r/t to the nozzle position, ignoring the probe offset.

Reply:
Why though? The user doesn't need to calculate the position with regards to the probe when it can be easily calculated from the `NOZZLE_TO_PROBE_OFFSET`
So if the user changes the probe mount they don't need to recompile the firmware with the new calculated and correct `POWER_LOSS_ZHOME_POS`, all they have to do is set the new probe offset with `M851`

Reply:
The idea is to set a position that is safe for the physical gantry (which doesn't change with the probe offset) to avoid the print. The larger the nozzle-to-probe offset is in Y, the more the gantry will encroach on the print area, increasing the likelihood of a crash into the print. So, that's my devil's advocate argument for keeping it that way. However, the `Z_SAFE_HOMING` point is probe-relative when a probe exists, so it would be consistent to subtract the probe offset, as in…
```cpp
TERN_(HOMING_Z_WITH_PROBE, destination -= probe.offset_xy);
```
…so I will go ahead and make that change.